Parse/Vehicle.swift

```diff
@@ -65,6 +67,34 @@ struct Vehicle {
         }
     }
     
+    /**
+     Estimates the current position of this Vehicle based on the previous Updates received.
+     - Returns: A Point corresponding to the current estimated position for this Vehicle.
+     */
+    func estimateCurrentPosition() -> Point {
+        let feet_per_second = (last_update.speed / 60) * 5280 // TODO: Assumes speed is in mph; could be km/h
+        let distance = feet_per_second * secondsSince(update: last_update) // In feet
+        let start = Point(update: last_update)
+        var startIndex = 0
+        for i in 0..<last_update.route.points.count {
+            if last_update.route.points[i] == start {
+                startIndex = i
+                break
+            }
+        }
+        var elapsedDistance = 0.0
+        var endPoint = start
+        for i in startIndex..<last_update.route.points.count {
+            elapsedDistance += start.distanceFrom(p: last_update.route.points[i])
+            if elapsedDistance > distance {
+                // TODO: Determine intermediate points based on excess distance
+                endPoint = last_update.route.points[i]
+                break
+            }
+        }
+        return endPoint
+    }
+    
```

Comment:
logic seems weird, please explain the second for loop.

Reply:
Before the second loop begins, we calculate the distance that the shuttle would travel if it continued traveling at the same speed as in the last update. We also find a starting location, the point in the route that is the closest to the shuttle's actual location. Then, in the second loop, we loop from the starting location through all the points in the route that come after that. For each point, we add the distance between this point and the last point to the total elapsed distance, and if the elapsed distance is now greater than the distance the shuttle would have traveled, we predict that the shuttle will be at this point.

Reply:
Fine, should probably find a better way to snap the vehicle to the route, but ok for now.

Reply:
See commit [4d97a8a](
https://github.com/quuu/ShuttleTrackeriOS/pull/28/commits/4d97a8adbb274855a040ab3b4c8d847c53dd03d1), vehicles now store a closest_point_index variable that gets initialized whenever the vehicle receives an update to snap them to a point on the route.